esp/logging/loggingagent/modularlogagent/README.md

```diff
@@ -0,0 +1,380 @@
+
+At the time of this writing, the binding XML must be manually edited. Support for automated assembly of bindings from smaller, reusable property files may be provided in the future.
+
+## Sample Configurations
+
+These examples show only mandatory *agent integration* and key *modular* properties. The illustration of optional *agent integration* properties is beyond the scope of this document.
+
+Each example provides both XML and YAML sample markup for inline configurations. With limited exceptions, the samples represent external property files content used by split configurations. These exceptions are addressed on a case by case basis.
+
+### Default "no-logging" Agent
+
+Conditionally append log update data to trace log output. The condition to ensure output is an ESP process *logLevel* property value of at least *9*.
+
+#### XML
+
+Either:
+
+    <LogAgent name="default-no-logging" plugin="scappsmodularlogagent" UpdateLog="true"/>
+
+or:
+
+    <LogAgent name="default-no-logging" plugin="scappsmodularlogagent">
+      <UpdateLog/>
+    </LogAgent>
+
+When originating from *configmgr*, if the *environment.xml* configuration omits both *configuration* and *trace-priority-level* properties, the second form is generated in *esp.xml*. No external property file is used.
+
+#### YAML
+
+    LogAgent:
+      name: default-no-logging
+      plugin: scappsmodularlogagent
+      UpdateLog: true
+
+### Verbose "no-logging" Agent
+
+Conditionally append log update data to trace log output. The condition to ensure output is a *trace-priority-limit* property value of at least *90*. Unlike in the **Default "no-logging" Agent** example, this agent controls its own trace output independent of the ESP process and any other configured agent.
+
+#### XML
+
+Either:
+
+    <LogAgent name="verbose-no-logging" plugin="scappsmodularlogagent" trace-priority-limit="90" UpdateLog="true"/>
+
+Or:
+
+    <LogAgent name="verbose-no-logging" plugin="scappsmodularlogagent" trace-priority-limit="90">
+      <UpdateLog/>
+    </LogAgent>
+
+When originating from *configmgr*, if the *environment.xml* configuration omits the *configuration* property, the second form is generated in *esp.xml*. No external property file is used.
+
+#### YAML
+
+    LogAgent:
+      name: verbose-no-logging
+      plugin: scappsmodularlogagent
+      trace-priority-limit: 90
+      UpdateLog: true
+
+### Terse "no-logging" Agent
+
+Refine the output produced by the **Verbose "no-logging" Agent** example by quieting the **LogAgent** and **UpdateLog** modules, and only ensuring recording the **Target** module input. As the complexities of the default modules, especially **UpdateLog**, increases, this example could reduce potentially excessive trace output.
+
+#### XML
+
+    <LogAgent name="terse-no-logging" plugin="scappsmodularlogagent" trace-priority-limit="10">
+      <UpdateLog trace-priority-limit="90"/>
```

Comment:
I'm confused how this is quieting the UpdateLog module when the limit = 90?

Reply:
It is a simplistic example of overriding the log level. The idea is that one module, in this case the agent, can be silenced, while a contained module, in this case the update log, can be verbose. Maybe "terse" in the heading isn't the best terminology, but the example is trying to show that trace output can be focused on an individual module.

Reply:
Upon re-reading, I understand your confusion. Will add the Target element referred to in the description.